blockchain/chain.go

```diff
@@ -651,9 +673,13 @@ func (b *BlockChain) connectBlock(node *blockNode, block *bchutil.Block,
 		return err
 	}
 
-	// Prune fully spent entries and mark all entries in the view unmodified
-	// now that the modifications have been committed to the database.
-	view.commit()
+	// Commit all modifications made to the view into the utxo state.  This also
+	// prunes these changes from the view.
+	b.stateLock.Lock()
+	if err := b.utxoCache.Commit(view); err != nil {
+		log.Errorf("error committing block %s(%d) to utxo cache: %s", block.Hash(), block.Height(), err.Error())
+	}
+	b.stateLock.Unlock()
```

Comment:
`Commit()` seems like a relatively risky operation. Maybe it's worth putting this in a function so that the `Unlock()` can be deferred?

Reply:
or just cheat and bundle it with the flush below.

Reply:
I concur. Wrap it in a function that executes immediately, and then the defer will cover the error case.

Reply:
Actually, deleted my previous comment. This should be fine. If the commit fails then we still unlock. So technically this is fine since we don't return the err.